src/structures/idt.rs

```diff
@@ -593,17 +593,51 @@ impl<T> PartialEq for Entry<T> {
 }
 
 /// A handler function for an interrupt or an exception without error code.
+#[cfg(feature = "abi_x86_interrupt")]
 pub type HandlerFunc = extern "x86-interrupt" fn(InterruptStackFrame);
+/// This type is a dummy.
+///
+/// This type alias is not used for builds which do not use the `abi_x86_interrupt` feature.
+#[cfg(not(feature = "abi_x86_interrupt"))]
+pub type HandlerFunc = ();
```

Comment:
We should just make these `Func`s opaque types if the interrupt feature isn't enabled. Something like:
```rust
pub type HandlerFunc(());
```

This has two advantages:

1. These types are no longer constructable, allowing us to change them going forward in a backwards-compatible way.
2. The types are distinct, so the corresponding different kinds of `Entry`s are distinct, which they definitely need to be.

Reply:
You mean, 
```rust
pub struct HandlerFunc(());
```
?

Reply:
Whoops, yes that's what I meant